src/dpp/guild.cpp

```diff
@@ -229,13 +229,20 @@ std::string guild_member::build_json(bool with_id) const {
 			j["communication_disabled_until"] = json::value_t::null;
 		}
 	}
-	if (!this->nickname.empty())
-		j["nick"] = this->nickname;
+	
+	if (!this->nickname.empty()) {
+		j["nick"] = json::value_t::null;
+	} else {
+		j["nick"] = find_user(user_id)->username;
```

Comment:
This should set the nickname from the passed member object, not from cache

Reply:
i have updated the new one